### ISSUE TITLE
"Direction Matters" for performance?

### ISSUE THREAD

**Issue by schwab** [RedisLabsModules/RedisGraph#291]
Consider this code which performs the same query twice, the only difference being the order in which the relationship is expressed. (-[]-> vs <-[]-)  The second query gives the same results, but it takes it 100X longer to do so (15.5 sec vs 147msec).  
```
%%time
import sys
from IPython.display import display, HTML
from redis_graph_provider import RedisGraphProvider
rgp = RedisGraphProvider()

g = rgp.get_graph("graph:x" need_master=False)
q_agg_1 = "MATCH (f:fact)-[:in]->(a:dim_patient) RETURN count(a)"
q_agg_2 = "MATCH (a:dim_patient)<-[:in]-(f:fact) RETURN count(a)"
%timeit rs = g.query(q_agg_1)
%timeit rs1 = g.query(q_agg_2)
print(rs.result_set)
print(rs1.result_set)
```
```
147 ms ± 1.51 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
15.5 s ± 229 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
[[b'count(a)'], [b'31257.000000']]
[[b'count(a)'], [b'31257.000000']]
CPU times: user 25.8 ms, sys: 3.23 ms, total: 29 ms
Wall time: 2min 15s
```

For context, the entire graph has about 600,000 links and consumes about 300MB of RAM. All the nodes are linked to from fact nodes in this case and non point at it. That said, I can see how size would matter, but I didn't expect direction of the query to have such a profound affect on the performance.  If this a problem in the way queries are processed, or do we need to find ways to work around this performance problem?

NOTE, this same behavior happens on the terminal with the redis-cli, so it's not a python issue. 

```
127.0.0.1:6379> GRAPH.QUERY graph:periop "MATCH (f:fact_periop)-[:in]->(a:dim_patient)  RETURN count(f) as count"
1) 1) 1) "count"
   2) 1) "31257.000000"
2) 1) "Query internal execution time: 165.151130 milliseconds"
127.0.0.1:6379> GRAPH.QUERY graph:periop "MATCH (a:dim_patient)<-[:in]-(f:fact_periop)  RETURN count(f) as count"
1) 1) 1) "count"
   2) 1) "31257.000000"
2) 1) "Query internal execution time: 14934.420148 milliseconds"
(14.94s)
127.0.0.1:6379> GRAPH.EXPLAIN graph:periop "MATCH (f:fact_periop)-[:in]->(a:dim_patient)  RETURN count(f) as count"
"Aggregate\n    Conditional Traverse\n        Node By Label Scan\n"
127.0.0.1:6379> GRAPH.EXPLAIN graph:periop "MATCH (a:dim_patient)<-[:in]-(f:fact_periop)  RETURN count(f) as count"
"Aggregate\n    Conditional Traverse\n        Node By Label Scan\n"
```

**Comment by swilly22** [comment]
@schwab thank you for pointing this one out,

Indeed direction in the queries you've provided shouldn't matter performance, behind the scene what happens in the latter query: `MATCH (a:dim_patient)<-[:in]-(f:fact) RETURN count(a)` is an unnecessary matrix transpose.

I believe this is an easy fix, will update here once implemented.

**Comment by jeffreylovitz** [comment]
Resolved in v1.0.12 - @schwab, please comment if you encounter further issues!
